File: image-store/src/main/java/com/expedia/Application.java
package com.expedia;

import com.expedia.cli.ImageStoreOptions;
import com.expedia.cli.CommandLineRunner;

public class Application {

    public static void main(String[] args) {
        try {
            CommandLineRunner.of(new ImageStoreOptions()).run(args);
        } catch (Exception e) {
            System.out.println("An error occurred: " + e.getMessage());
        }
    }
}

File: dynamic-programming/vending-machine/src/main/kotlin/Application.kt
import cli.VendingHelpFormatter
import cli.VendingOptions
import com.xenomachina.argparser.ArgParser
import com.xenomachina.argparser.SystemExitException
import service.VendingMachine
import java.io.StringWriter

fun main(args: Array<String>) {
    try {
        val parser = ArgParser(args, helpFormatter = VendingHelpFormatter())
        VendingOptions(parser).run {
            val vendingMachine = VendingMachine(inventory = inventory)
            val amount = readLine()!!.toInt()
            vendingMachine.dispense(amount)
        }
    } catch (e: SystemExitException) {
        println(StringWriter().apply {
            e.printUserMessage(this, "vending-machine", 60)
        })
    } catch (e: Exception) {
        println("An error occurred: ${e.message}")
    }
}

